### DISCUSSION TITLE
ShieldOAuthConfig.php in App/Config folder

### DISCUSSION THREAD

**Discussion by AlbertoMoricca** [datamweb/shield-oauth#23] [Q&A] [answered]
Hi,

I installed **shield-oauth** library with composer. The documentation says that the Oauth keys need to be set in `App/Config/ShieldOAuthConfig.php` file. I have no this file, then I think that the documentation refers to the file `vendor/datamweb/shield-oauth/src/Config//ShieldOAuthConfig.php`.

I prefer to not edit the file under vendor. Is it possible to copy the file under `App/Config` and use it (overriding the default file under vendor) without changing the code of the library?

Thanks,
Alberto

**Comment by datamweb** [comment]
**Hey Alberto**,

In fact, as mentioned in the documentation, I intended to cover this issue from the beginning, but it seems that I did not pay attention to this issue during the development process.
So you are right, don't set it in `vendor`.
I will cover this, but I don't think I will have free time to work on it for the next two weeks. If by that time no PR is sent to this by any person, I will do it that way.

**Comment by AlbertoMoricca** [comment]
Hi,

Ok, thanks for the clarification :)

**Reply by datamweb** [reply]
Hello @AlbertoMoricca.
I had a cursory review of the code, it seems the necessary stuff has already been done.

Please follow the steps below:

1. Copy file `vendor\datamweb\shield-oauth\src\Config\ShieldOAuthConfig.php` and paste  it in path `app\Config\ShieldOAuthConfig.php`.
2. Edit file `app\Config\ShieldOAuthConfig.php` and make the following changes:
```diff
-namespace Datamweb\ShieldOAuth\Config;
+namespace Config;

-use CodeIgniter\Config\BaseConfig;
+use Datamweb\ShieldOAuth\Config\ShieldOAuthConfig as OAuthConfig;

-class ShieldOAuthConfig extends BaseConfig
+class ShieldOAuthConfig extends OAuthConfig
```
3. Setting the keys in the file `app\Config\ShieldOAuthConfig.php`.

Please follow the steps above and let me know if it worked well, or send a PR.

**Comment by datamweb** [comment, marked as answer]
**I sent PR #24 .** 

Update the package with command :
```console
composer require datamweb/shield-oauth:dev-develop
```
Then run the following command to create file `app\Config\ShieldOAuthConfig.php`:
```console
php spark make:oauthconfig
```

**Reply by AlbertoMoricca** [reply]
Hi,

sorry for the no answer, but i had no receive a notification about your answer.

I tried your solution and the configuration works.

Unfortunately, I can't complete the login because the fetchUserInfoWithToken() return null.

I'm trying to understand the issue. I will open another topic because the issue it seems not relatives to the configuration file.

Thanks for your help!

Alberto
